src/main/java/gregtech/api/util/OverlayedItemHandler.java

```diff
@@ -100,22 +100,24 @@ public int insertStackedItemStackKey(ItemStackKey key, int amountToInsert) {
     }
 
     private static class OverlayedItemHandlerSlot {
-        private ItemStackKey itemStackKey = null;
-        private int count = 0;
-        private int slotLimit = 0;
+        private ItemStack itemStack;
+        private int count;
+        private int slotLimit;
 
-        OverlayedItemHandlerSlot(ItemStack stackToMirror, int slotLimit) {
+        protected OverlayedItemHandlerSlot(@Nonnull ItemStack stackToMirror, int slotLimit) {
             if (!stackToMirror.isEmpty()) {
-                this.itemStackKey = KeySharedStack.getRegisteredStack(stackToMirror);
+                this.itemStack = stackToMirror.copy();
                 this.count = stackToMirror.getCount();
-                this.slotLimit = Math.min(itemStackKey.getMaxStackSize(), slotLimit);
+                this.slotLimit = Math.min(itemStack.getMaxStackSize(), slotLimit);
             } else {
+                this.itemStack = ItemStack.EMPTY;
+                this.count = 0;
```

Comment:
Why the move to setting values here instead of just initializing the variables when they are declared?